Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource_printer.go

```diff
@@ -320,10 +320,12 @@ type handlerEntry struct {
 type PrintOptions struct {
 	NoHeaders          bool
 	WithNamespace      bool
+	WithKind           bool
 	Wide               bool
 	ShowAll            bool
 	ShowLabels         bool
 	AbsoluteTimestamps bool
+	KindName           string
```

Comment:
> Why does a printer need to know about a kind name?! Isn't the printer generic and able to print resources of different kinds?!

I agree with this. At least for kube resources, I pass the resource alias (full name if no alias) to a printer method here: https://github.com/openshift/origin/pull/9334/commits/259f9890c88b2dafaa7d95cd9ed1ae9c343ad384#diff-85d15d1be2a3707c4cc1b4e658dff84aR363

Perhaps making a separate method, maybe something like `SetResourceAlias`, would work better. That way, I can call it from `get.go` for kube resources, and from our `factory.go` Printer method here: https://github.com/openshift/origin/pull/9334/commits/8bf8c5532adbd09ed337fc108146c7e5cc6568e8#diff-0073015bc0e88b3269344b5a87d471dbR380 for our resources